### PR TITLE
Add support for passwords with quotation marks, backslashes and tabs when configuring cloud provider and CSI

### LINKED ISSUE #892: INI file escaping required for CSI configuration
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
CSI configuration fails if certain, typically escaped INI file characters are used.

**What did you expect to happen:**
Can use acceptable vcenter passwords

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-vsphere version: 0.6.3
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by randomvariable**
**What this PR does / why we need it**:
Corrects character escaping in ini file generation for CSI and the cloud provider. 
Additionally verifies password encoding for Active Directory in tests.
Deleted the old encoding functions from v1alpha2 which are not used for the API contract.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: 
Fixes #892 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Add support for passwords with quotation marks, backslashes and tabs when configuring cloud provider and CSI
```

### CODE REVIEW

**Comment by yastij** [comment]
@randomvariable - seems like you need to run `make generate`

**Comment by randomvariable** [comment]
There's some other bits to sort out. Fixing.

**Comment by randomvariable** [comment]
Ready to merge

File: api/v1alpha3/cloudprovider_encoding_test.go [hunk 1]
```diff
@@ -149,18 +150,18 @@ folder = kubernetes
...
-port = 443
-datacenters = us-west
+secret-name = "vccreds"
+secret-namespace = "kube-system"
+port = "443"
+datacenters = "us-west"
 
 [VirtualCenter "0.0.0.0"]
-password = password
+password = "password"
```
Comment (yastij) [inline api/v1alpha3/cloudprovider_encoding_test.go:159]: can you add a test case for a password/username with a special char ?

File: api/v1alpha2/cloudprovider_encoding.go [hunk 2]
```diff
@@ -1,248 +0,0 @@
-/*
-Copyright 2019 The Kubernetes Authors.
```
[the remark this answers is not in the document]
Comment (randomvariable) [reply api/v1alpha2/cloudprovider_encoding.go:2]: It's dead code. The encoding is only used in the controller in the v3 variant, it's not part of the API. There's a case to be made to move these encoding functions out of the API, as is done in CAPA with the cluster scope / machine scope package.

[the remark this answers is not in the document]
Comment (yastij) [reply api/v1alpha2/cloudprovider_encoding.go:2]: this is used in v1alpha2 see https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/release-0.5/controllers/vspherecluster_controller.go#L529

let's leave it there, If someone wants the fix, they'll need to upgrade to `v0.6`

[the remark this answers is not in the document]
Comment (randomvariable) [reply api/v1alpha2/cloudprovider_encoding.go:2]: I'm merging into master. If this fix is going to be backported, the fixes would need to go into the old directory anyway.

**Comment by ncdc** [comment]
Is this to fix #892?

**Comment by randomvariable** [comment]
yup, updated desc

**Pull request merged**